src/cluster.c

```diff
@@ -5031,6 +5105,16 @@ NULL
     }
 }
 
+void removeChannelsInSlot(unsigned int slot) {
+    unsigned int channelcount = countChannelsInSlot(slot);
+    if (channelcount != 0) {
+        robj **channels = zmalloc(sizeof(robj*)*channelcount);
+        getChannelsInSlot(slot,channels);
+        pubsubUnsubscribeLocalChannels(channels,channelcount);
+        zfree(channels);
+    }
```

Comment:
purely stylistic, but I think it's usually easier to follow code that is written where validation happens and returns if it fails instead of having if/else blocks. 
```suggestion
    unsigned int channelcount = countChannelsInSlot(slot);
    if (channelcount == 0) return;
    
    robj **channels = zmalloc(sizeof(robj*)*channelcount);
    getChannelsInSlot(slot,channels);
    pubsubUnsubscribeLocalChannels(channels,channelcount);
    zfree(channels);
```